kura/test/org.eclipse.kura.core.keystore.test/src/test/java/org/eclipse/kura/core/keystore/FilesystemKeystoreServiceImplTest.java

```diff
@@ -119,6 +121,43 @@ public void testGetKeyStore() throws KuraException, KeyStoreException {
         assertEquals(Collections.list(this.store.aliases()), Collections.list(keystore.aliases()));
     }
 
+    @Test
+    public void testCheckKeystoreFileCreation() throws KuraException, KeyStoreException {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(KEY_KEYSTORE_PATH, NEW_STORE_PATH);
+
+        CryptoService cryptoService = mock(CryptoService.class);
+        ComponentContext componentContext = mock(ComponentContext.class);
+
+        FilesystemKeystoreServiceImpl keystoreService = new FilesystemKeystoreServiceImpl();
+        keystoreService.setEventAdmin(mock(EventAdmin.class));
+        keystoreService.setCryptoService(cryptoService);
+        keystoreService.activate(componentContext, properties);
+
+        assertTrue(new File(NEW_STORE_PATH).exists());
+
+    }
+
+    @Test
+    public void testCheckKeystoreFileCreationWithUpdate() throws KuraException, KeyStoreException {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(KEY_KEYSTORE_PATH, NEW_STORE_PATH);
+
+        CryptoService cryptoService = mock(CryptoService.class);
+        ComponentContext componentContext = mock(ComponentContext.class);
+
+        FilesystemKeystoreServiceImpl keystoreService = new FilesystemKeystoreServiceImpl();
+        keystoreService.setEventAdmin(mock(EventAdmin.class));
+        keystoreService.setCryptoService(cryptoService);
+        keystoreService.activate(componentContext, properties);
+
+        properties.put(KEY_KEYSTORE_PATH, UPDATED_NEW_STORE_PATH);
+
+        keystoreService.updated(properties);
+
+        assertTrue(new File(NEW_STORE_PATH).exists());
```

Comment:
Is there something ensuring filesystem cleanup after this test is ran? If not, the file could persist on the filesystem and invalidate the result of the following tests.